pkg/starkapi/forms_api.go

```diff
@@ -80,30 +86,53 @@ func (formsApi *FormsApi) GetControlByName(name string) (domain.FormControl, err
 	return control, nil
 }
 
-func (formsApi *FormsApi) CreateControlOnRef(control domain.FormControl) (domain.FormControl, error) {
-	err := control.Validate()
+func (formsApi *FormsApi) CreateControlOnRef(formControlName string, ref string, value string) (domain.FormControlRef, error) {
```

Comment:
Refactor to take in a formControl struct vs just the name. https://en.wikipedia.org/wiki/Single-responsibility_principle. the current method has multiple responsibilities. 1) create for control ref.  2) look up a formControl. in order to use this method a user would have to look up a formControl or a list of form controls to get a valid name